### ISSUE TITLE
Non-ad content being marked as "Ad"

### ISSUE THREAD

**Issue by rand0me** [chromium/chromium-ads-detection#60]
Hey guys, it looks like the Ad Tagging mechanism has a flaw that makes our users suffer from blocking non-ad content (web games) by Heavy Ad Intervention:

https://www.arkadium.com/games/word-wipe

Indeed, game iframe has `ima3.js` in its creation call-stack - but this happens because we run pre-roll video ad prior to render it. We rely on `ima3.js` events - such as `skip`, `complete`, `error` to understand when the video ad has finished. And unfortunately, most of our games are more than 4MB, this is why we have this issue happening very often.

Could you please take a look if this is something that could be fixed here? Would we be able to add an exclusion (though we have >200 sites, such as games.dailymail.co.uk with our games and same rendering method).

I'd be very thankful for any information or advice how to avoid getting a game container tagged as "Ad" on our websites.

Thank you!

**Comment by jkarlin** [comment]
Right, ima3.js in the stack of the game iframe creation would cause Chrome to think that your game iframe is an ad frame. Can you create the game iframe independently and hide it until after the ad completes?

**Comment by johnivdel** [comment]
Thanks @rand0me for the update. Looking at the site now, it looks like the frame is not being tagged as an ad for me. Does the repro link above still work for you?

My expectation is that (2) should work, navigating the frame with ad script in the stack shouldn't re-tag it.  Is there any other ad script in the stack when you try and create the iframe before the video ad?

You could also try setting an initial src on the frame like: "...arkadiumhosted.com/empty_frame.html" and see if that causes a difference.
